internal/pkg/manifest/transform.go

```diff
@@ -0,0 +1,208 @@
+		case "NetworkConfig":
+			dstElem := dst.Interface().(*NetworkConfig)
+			srcElem := src.Elem().Interface().(NetworkConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "vpcConfig":
+			dstElem := dst.Interface().(*vpcConfig)
+			srcElem := src.Elem().Interface().(vpcConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "Logging":
+			dstElem := dst.Interface().(*Logging)
+			srcElem := src.Elem().Interface().(Logging)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "Storage":
+			dstElem := dst.Interface().(*Storage)
+			srcElem := src.Elem().Interface().(Storage)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "Alias":
+			dstElem := dst.Interface().(*Alias)
+			srcElem := src.Elem().Interface().(Alias)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "SidecarConfig":
+			dstElem := dst.Interface().(*SidecarConfig)
+			srcElem := src.Elem().Interface().(SidecarConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "SubscribeConfig":
+			dstElem := dst.Interface().(*SubscribeConfig)
+			srcElem := src.Elem().Interface().(SubscribeConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "SQSQueue":
+			dstElem := dst.Interface().(*SQSQueue)
+			srcElem := src.Elem().Interface().(SQSQueue)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "EFSConfigOrBool":
+			dstElem := dst.Interface().(*EFSConfigOrBool)
+			srcElem := src.Elem().Interface().(EFSConfigOrBool)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "AuthorizationConfig":
+			dstElem := dst.Interface().(*AuthorizationConfig)
+			srcElem := src.Elem().Interface().(AuthorizationConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		}
+
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func transformPBasic() func(dst, src reflect.Value) error {
+	return func(dst, src reflect.Value) error {
+		if src.IsNil() {
+			return nil
+		}
+		dst.Set(src)
+		return nil
+	}
+}
+
+// transformImage implements customized merge logic for Image field of manifest.
+// It merges `DockerLabels` and `DependsOn` in the default manager (i.e. with configurations mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+// And then overrides both `Build` and `Location` fields at the same time with the src values, given that they are non-empty themselves.
+func transformImage() func(dst, src reflect.Value) error {
```

Comment:
I can understand that `volumes` is a unique one because it is a map of struct (correct me if I am wrong) but why do we need customized merge logic for `Image` 🤔 If we don't need, I think ultimately the only special cases we might want to deal with in our manifest can be abstracted as 1) map of structs 2) slice of structs (since we are very unlikely to have the other composite types such as nested map or slice).

Reply:
We need it because `build` and `location` are mutually exclusive. One being set requires the other being unset. This can't be done with any of the existing transformers logic